src/warzoneconfig.cpp

```diff
@@ -718,3 +724,34 @@ bool war_getAllowVulkanImplicitLayers()
 {
 	return warGlobs.allowVulkanImplicitLayers;
 }
+
+void war_setHostConnectionProvider(ConnectionProviderType pt)
+{
+	warGlobs.hostProviderType = pt;
+}
+
+ConnectionProviderType war_getHostConnectionProvider()
+{
+	return warGlobs.hostProviderType;
+}
+
+bool net_backend_from_str(const char* str, ConnectionProviderType& pt)
+{
+	if (strcasecmp(str, "tcp") == 0)
+	{
+		pt = ConnectionProviderType::TCP_DIRECT;
+		return true;
+	}
+	return false;
+}
+
+std::string to_string(ConnectionProviderType pt)
+{
+	switch (pt)
+	{
+	case ConnectionProviderType::TCP_DIRECT:
+		return "tcp";
+	default:
+		throw std::runtime_error("Invalid connection provider type enumeration value");
+	}
```

Comment:
```suggestion
	}
	ASSERT_OR_RETURN("", false, "Invalid connection provider type enumeration value: %d", static_cast<int>(pt)); // silence GCC warning
```

Let’s ensure a compile-time error if the ConnectionProviderType enum is extended, and avoid throwing an exception in the bit that’s required to avoid a GCC warning.

Reply:
OK, will fix that.